framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -706,7 +723,11 @@ private static void printClass(
             printWriter.println();
             for (Map.Entry<String, AMethod> methodEntry : aClass.getMethods().entrySet()) {
                 printMethodDeclaration(
-                        methodEntry.getValue(), innermostClassname, printWriter, indentLevel);
+                        methodEntry.getValue(),
+                        innermostClassname,
+                        printWriter,
+                        indentLevel,
+                        checker.getTypeFactory());
```

Comment:
This does pass the AnnotatedTypeFactory.

Are you suggesting that the containing method printClass could have its formal parameter list changed, so it takes an AnnotatedTypeFactory rather than a checker?  The changes in #4025 require the checker rather than the type factory.